.github/workflows/production-release.yml

```diff
@@ -0,0 +1,83 @@
+name: Production Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Release version (e.g., v2.0.3)'
+        required: true
+        type: string
+      dry_run:
+        description: 'If true, does a dry run of the production workflow'
+        required: false
+        type: boolean
+
+run-name: ${{ inputs.dry_run && '[DRY RUN] ' || '' }}Release ${{ inputs.version }} by @${{ github.actor }}
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  contents: write
```

Comment:
Should this be default of "read" and then each job defines "write" if necessary?